ugvc/cnv/normalize_reads_count.R

```diff
@@ -0,0 +1,314 @@
+# Copyright 2022 Ultima Genomics Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+# DESCRIPTION
+#    whole genome reads count normalization for a given cohort.
+#    reads count normalization function is based on cn.mops package documented here : https://academic.oup.com/nar/article/40/9/e69/1136601
+# CHANGELOG in reverse chronological order
+
+suppressPackageStartupMessages(library(cn.mops))
+suppressPackageStartupMessages(library(magrittr))
+suppressPackageStartupMessages(library(argparse))
+
+
+.statmod <- function(x,na.rm=FALSE) {
+	if (na.rm){
+		z <- table(as.vector(x[!is.na(x)]))
+		r <- names(z)[z == max(z)]
+		return(as.numeric(r)[1])
+	} else {
+		if (any(is.na(x))){return(NA)
+		} else {
+			z <- table(as.vector(x))
+			r <- names(z)[z == max(z)]
+			return(as.numeric(r)[1])
+		}
+	}
+}
+
+GetVarianceSum <- function(X,in_cutoff){
+  left_dist = X[X<=in_cutoff]
+  right_dist = X[X>in_cutoff]
+  left_variance = var(left_dist)
+  right_variance = var(right_dist)
+  sum_var = left_variance+right_variance
+  return(sum_var)
+}
+
+#' @title whole Genome-wize Normalization of NGS data.
+#'
+#' @description Normalize quantitative NGS data in order to make counts comparable over
```

Comment:
Why don't we add this function to cn.mops actually?

Reply:
The main reason for not moving this script to cn.mops repo is that it does not have tests. We have a made a test for this particular script within current repo.

Reply:
Got it... sounds like we made something pretty ugly. You can merge it for now. Can you please leave jira issue to fix this? 
I think we should create a conda package for our version of cn.mops probably to make it more reasonable

Reply:
Agreed: https://ultimagen.atlassian.net/browse/BIOIN-1121